server/src/main/java/org/opensearch/index/remote/RemoteStoreUtils.java

```diff
@@ -146,4 +152,30 @@ static String longToCompositeBase64AndBinaryEncoding(long value, int len) {
         assert base64DecimalValue >= 0 && base64DecimalValue < 64;
         return URL_BASE64_CHARSET[base64DecimalValue] + binaryPart;
     }
+
+    /**
+     * Fetches segment and translog repository names from remote store node attributes
+     * @param discoveryNodes Current set of {@link DiscoveryNodes} in the cluster
+     * @return {@link Tuple} with segment repository name as first element and translog repository name as second element
+     */
+    public static Map<String, String> getRemoteStoreRepoName(DiscoveryNodes discoveryNodes) {
+        Optional<DiscoveryNode> remoteNode = discoveryNodes.getNodes()
+            .values()
+            .stream()
+            .filter(DiscoveryNode::isRemoteStoreNode)
+            .findFirst();
+        assert remoteNode.isPresent() : "Cannot fetch remote store repository names as no remote nodes are present in the cluster";
+        return remoteNode.get().getRemoteStoreRepoNames();
+    }
+
+    /**
+     * Ensures that the current set of {@link DiscoveryNodes} contains at-least one node
+     * matching the {@link DiscoveryNode#isRemoteStoreNode()} predicate
+     *
+     * @param discoveryNodes Current set of {@link DiscoveryNodes}
+     * @return <code>true</code> if there is at-least one remote store enabled node, <code>false</code> otherwise
+     */
+    public static boolean hasAtLeastOneRemoteNode(DiscoveryNodes discoveryNodes) {
+        return discoveryNodes.getNodes().values().stream().anyMatch(DiscoveryNode::isRemoteStoreNode);
+    }
```

Comment:
we can remove this . Even if there is no remote node, our code changes doesn't affect the cluster in any way.

Reply:
Addressed